src/lib/services/historyService.ts

```diff
@@ -44,9 +49,19 @@ export async function findHistory(historyProps: HistoryProps): Promise<Array<His
 
   const labwareMap: Map<number, LabwareFieldsFragment> = new Map();
   const sampleMap: Map<number, SampleFieldsFragment> = new Map();
+  const samplePositionMapByOpId: Map<number, Map<number, SamplePositionFieldsFragment>> = new Map();
 
   history.labware.forEach((lw) => labwareMap.set(lw.id, lw));
   history.samples.forEach((sample) => sampleMap.set(sample.id, sample));
+  history.samplePositionResults.forEach((samplePosition) => {
```

Comment:
samplePositionResults are not associated with history data .Better to use  _samplePositions_ query on core side

Reply:
I updated the core code accordingly, so the samplePositions is part of the History object, unless you mean this is not logically correct, and we need to perform two separate query